app/controllers/declaration/SealController.scala

```diff
@@ -42,78 +44,124 @@ class SealController @Inject()(
   errorHandler: ErrorHandler,
   override val exportsCacheService: ExportsCacheService,
   mcc: MessagesControllerComponents,
-  sealPage: seal
+  addPage: seal_add,
+  removePage: seal_remove,
+  summaryPage: seal_summary
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable {
 
-  def displayPage(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val declaration = request.cacheModel
-    Ok(sealPage(mode, form(), declaration.seals, declaration.transportDetails.exists(_.container)))
+  def displayAddSeal(mode: Mode, containerId: String): Action[AnyContent] = (authenticate andThen journeyType) {
+    implicit request =>
+      Ok(addPage(mode, Seal.form(), containerId))
   }
 
-  def submitForm(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val actionTypeOpt = FormAction.bindFromRequest()
-    val boundForm = form().bindFromRequest()
-
-    val data = request.cacheModel
-    val seals = data.seals
-    val hasContainers: Boolean = data.transportDetails.fold(false)(_.container)
-    actionTypeOpt match {
-      case Add                             => addSeal(mode, boundForm, sealsAllowed, seals)
-      case Remove(ids)                     => removeSeal(mode, boundForm, seals, hasContainers, ids)
-      case SaveAndContinue | SaveAndReturn => saveSeal(mode, boundForm, sealsAllowed, seals)
-      case _                               => errorHandler.displayErrorPage()
-    }
-  }
+  def submitAddSeal(mode: Mode, containerId: String): Action[AnyContent] = (authenticate andThen journeyType).async {
+    implicit request =>
+      val boundForm = Seal.form().bindFromRequest()
 
-  private def saveSeal(mode: Mode, boundForm: Form[Seal], elementLimit: Int, cachedSeals: Seq[Seal])(
-    implicit request: JourneyRequest[AnyContent]
-  ): Future[Result] =
-    saveAndContinue(boundForm, cachedSeals, isMandatory = false, elementLimit).fold(
-      formWithErrors => Future.successful(badRequest(mode, formWithErrors, cachedSeals)),
-      updatedCache =>
-        if (updatedCache != cachedSeals) {
-          updateCache(updatedCache).map { _ =>
-            navigator.continueTo(routes.SummaryController.displayPage(Mode.Normal))
-          }
-        } else Future.successful(navigator.continueTo(routes.SummaryController.displayPage(Mode.Normal)))
-    )
+      request.cacheModel.containers.find(_.id == containerId) match {
```

Comment:
can we move this find to container or even ExportDeclaration ?

Reply:
I had already created the method (after your last comment) - just forgot to use it!